src/components/DistanceRequest.js

```diff
@@ -65,6 +80,8 @@ function DistanceRequest({transactionID, transaction, translate}) {
     const halfMenuItemHeight = Math.floor(variables.baseMenuItemHeight / 2);
     const scrollContainerMaxHeight = variables.baseMenuItemHeight * MAX_WAYPOINTS_TO_DISPLAY + halfMenuItemHeight;
 
+    useEffect(() => MapboxToken.init(), []);
```

Comment:
No cleanup needed on unmount?

Reply:
@neil-marcellini I don't see `init` function defined in `src/libs/actions/MapboxToken.js` doesn't return a clean up function to call to remove listeners to `Onyx` and `AppState`.

Should I edit the function so that I have a cleanup function to call inside `useEffect` where the `init` function is invoked?

Reply:
@neil-marcellini so for this, we have two options.

Option 1: We can keep the current code. With this approach, we have three listeners/subscriptions that are always on once `Mapbox.init` function is called once. For now, we can keep the code as is, but we can clean up that is described in the option 2 it as a follow up issue because this PR is blocking other PRs.

- Subscription to `ONYXKEYS.SESSION` key (we're subscribing to `SESSION` to get the user's `authToken`, which is used to check the user's login state). We need to know the user's login state in the current implementation.
- Subscription to `ONYXKEYS.MAPBOX_ACCESS_TOKEN`
- Subscription to `AppState`

Option2: We can do refactoring right now, which probably involves using a hook. We only need the secret token when we display `MapView`. If we take that into account, we probably don't have to listen to `AppState` and also don't have to set a timeout for the token (we can instead store the timestamp for when a fresh token was fetched last time and refetch it if more than 25 min has passed since that time). We can just set the `token` undefined until the fresh token is fetched from the backend.

Let me know what you think @neil-marcellini imo, I think option 1 is better as this PR is blocking other PRs

Reply:
Ok yes good points, let's do the subscription clean up in a follow up. I'll create an issue.

> we probably don't have to listen to AppState

I think we need to because if you close the App on the distance request page and open it up 1 hour later the component doesn't mount again.

We talked about this in person and agreed that we still need the expiration timeout.

Reply:
- [ ] https://github.com/Expensify/App/issues/24612